### PR TITLE
Add utility files for building external plugins

### PR DESCRIPTION
**Pull request by cmcfarlen**
This adds both a pkg-config and cmake find-package script to help users build external plugins.

### CODE REVIEW

File: doc/developer-guide/plugins/building-plugins.en.rst [hunk 1]
```diff
@@ -22,3 +22,58 @@
...
+libraries, those can be linked with the :func:`target_link_libraries` cmake
+function.
+
+After the plugin target is created, a verify test target can be created.  This
+will add a cmake test declaration and at test time, will run traffic_server in
+its verify plugin mode.  There are two verify functions provided,
+:func:`verify_remap_plugin` and :func:`verify_global_plugin`.  Use the relevant
+function for your plugin type.  The test will fail if either your plugin does
+not define the required init function (:func:`TSRemapInit` or :func:`TSPluginInit`),
+or if your plugin calls any undefined symbols.
```
Comment (ywkaras) [inline doc/developer-guide/plugins/building-plugins.en.rst:62]: What about plugins that are both remap and global plugins?

File: ts.pc.in [hunk 2]
```diff
@@ -0,0 +1,11 @@
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=@CMAKE_INSTALL_FULL_LIBEXECDIR@
+includedir=@CMAKE_INSTALL_FULL_INCLUDEDIR@
+libdir=@CMAKE_INSTALL_FULL_LIBDIR@
+
+Name: trafficserver
+Description: Apache Traffic Server
+Version: @TS_VERSION_STRING@
+Requires:
+Libs: -L${libdir} -ltsapi
```
Comment (ywkaras) [inline ts.pc.in:10]: See https://github.com/apache/trafficserver/issues/11032

File: doc/developer-guide/plugins/building-plugins.en.rst (hunk 1 above)
Comment (cmcfarlen) [reply doc/developer-guide/plugins/building-plugins.en.rst:62]: I suppose you could request both tests.  This isn't a separate command for traffic_server afaik.

**Comment by ywkaras** [comment]
@dragon512 do you have time to look at this?

File: doc/developer-guide/plugins/building-plugins.en.rst [hunk 3]
```diff
@@ -22,3 +22,58 @@
...
+======================
+
+.. code-block:: text
+
+   cmake_minimum_required(VERSION 3.20..3.27)
+   project(ats-plugin-test)
+
+   find_package(tsapi REQUIRED)
+
+   enable_testing()
```
Comment (ywkaras) [inline doc/developer-guide/plugins/building-plugins.en.rst:43]: Would I know what this does if I weren't cmake-impaired?

File: doc/developer-guide/plugins/building-plugins.en.rst (hunk 1 above)
Comment (ywkaras) [reply doc/developer-guide/plugins/building-plugins.en.rst:62]: So should you mention that in the documentation?

**Review by ywkaras — APPROVED** [review]

**Pull request merged**